File: library/src/main/java/com/yy/codex/uikit/CGShadow.java
package com.yy.codex.uikit;

/**
 * Created by adi on 17/1/17.
 */

public class CGShadow {

    public final double radius;
    public final double offsetX;
    public final double offsetY;
    public final UIColor color;

    public CGShadow(double radius, double offsetX, double offsetY, UIColor color) {
        this.radius = radius;
        this.offsetX = offsetX;
        this.offsetY = offsetY;
        this.color = color;
    }

    public CGShadow setRadius(double radius) {
        return new CGShadow(radius, this.offsetX, this.offsetY, this.color);
    }

    public CGShadow setOffsetX(double offsetX){
        return new CGShadow(this.radius, offsetX, this.offsetY, this.color);
    }

    public CGShadow setOffsetY(double offsetY){
        return new CGShadow(this.radius, this.offsetX, offsetY, this.color);
    }

    public CGShadow setOffsetX(UIColor color){
        return new CGShadow(this.radius, offsetX, this.offsetY, color);
    }

}

File: library/src/main/java/com/yy/codex/uikit/CGShadow.kt
package com.yy.codex.uikit

/**
 * Created by adi on 17/1/17.
 */

class CGShadow(val radius: Double, val offsetX: Double, val offsetY: Double, val color: UIColor) {

    fun setRadius(radius: Double): CGShadow {
        return CGShadow(radius, this.offsetX, this.offsetY, this.color)
    }

    fun setOffsetX(offsetX: Double): CGShadow {
        return CGShadow(this.radius, offsetX, this.offsetY, this.color)
    }

    fun setOffsetY(offsetY: Double): CGShadow {
        return CGShadow(this.radius, this.offsetX, offsetY, this.color)
    }

    fun setOffsetX(color: UIColor): CGShadow {
        return CGShadow(this.radius, offsetX, this.offsetY, color)
    }

    override fun equals(other: Any?): Boolean {
        var other = other as? CGShadow ?: return false
        return Math.abs(radius - other.radius) < 0.01 && Math.abs(offsetX - other.offsetX) < 0.01 && Math.abs(offsetY - other.offsetY) < 0.01 && color == other.color
    }

}
